util/internal/integrationtest/backend/elasticsearch/src/main/java/org/hibernate/search/util/impl/integrationtest/backend/elasticsearch/extension/TestElasticsearchClient.java

```diff
@@ -48,26 +48,34 @@
 import org.hibernate.search.util.impl.integrationtest.backend.elasticsearch.dialect.ElasticsearchTestDialect;
 import org.hibernate.search.util.impl.integrationtest.common.TestConfigurationProvider;
 
-import org.junit.rules.TestRule;
-import org.junit.runner.Description;
-import org.junit.runners.model.Statement;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
 
 import com.google.gson.GsonBuilder;
 import com.google.gson.JsonArray;
 import com.google.gson.JsonElement;
 import com.google.gson.JsonObject;
 import com.google.gson.JsonParser;
 
-public class TestElasticsearchClient implements TestRule, Closeable {
+public class TestElasticsearchClient implements BeforeEachCallback, AfterEachCallback, Closeable {
 
 	private final ElasticsearchTestDialect dialect = ElasticsearchTestDialect.get();
 
+	private TestConfigurationProvider configurationProvider = new TestConfigurationProvider();
 	private ThreadPoolProviderImpl threadPoolProvider;
 	private ScheduledExecutorService timeoutExecutorService;
 	private ElasticsearchClientImplementor client;
 
 	private final List<URLEncodedString> createdIndicesNames = new ArrayList<>();
 
+	private TestElasticsearchClient() {
+	}
+
+	public static TestElasticsearchClient create() {
+		return new TestElasticsearchClient();
+	}
+
```

Comment:
😋 haha, yeah.... since this is "an extension" (it has some junit callbacks implemented) and as most of our extensions are using some static method to init itself in the test class, I wanted to make it more consistent and updated this one to follow that approach as well